src/R/Components/Impl/ConnectionManager/Connection.cs

```diff
@@ -14,21 +16,29 @@ public class Connection : ConnectionInfo, IConnection {
 
         public static Connection Create(ISecurityService securityService, string name, string path, string rCommandLineArguments, bool isUserCreated, bool fetchHostLoad) {
```

Comment:
At some point it may be worth actually splitting class into base connection, `UrlConnection`, `ContainerConnection` and `LocalConnection` and have `Create` as factory since we have several `IsABC` properties which is generally a bad form.